packages/tools/replay-tool/src/replayMessages.ts

```diff
@@ -777,3 +814,69 @@ export class ReplayTool {
         }
     }
 }
+
+async function assertDdsEqual(d1: ITreeEntry | undefined, d2: ITreeEntry | undefined): Promise<void> {
+    if (d1?.type !== TreeEntry.Tree || d2?.type !== TreeEntry.Tree) {
+        strict.deepStrictEqual(d1, d2);
+        return;
+    }
+
+    const attributes = d1.value.entries.find(
+        (entry) => entry.type === TreeEntry.Blob && entry.path === ".attributes",
+    );
+
+    const parsed: { type?: string; } = attributes?.type === TreeEntry.Blob
+                                            ? JSON.parse(attributes?.value.contents)
+                                            : {};
+
+    if (parsed.type !== SharedMatrixFactory.Type) {
+        assert(d1.value.entries.length === d2.value.entries.length, "");
+        for (let i = 0; i < d1.value.entries.length; i++) {
+            await assertDdsEqual(d1.value.entries[i], d2.value.entries[i]);
+        }
+
+        return;
+    }
+
+    const dataStoreRuntime = new MockFluidDataStoreRuntime();
+    dataStoreRuntime.local = true;
+    const deltaConnection = new MockEmptyDeltaConnection();
+
+    const objectStorage1 = MockStorage.createFromSummary(
+        convertToSummaryTreeWithStats(d1.value).summary,
+    );
+    const matrix1 = new SharedMatrix(
+        dataStoreRuntime as any,
+        "123",
```

Comment:
giving more suggestive ids here would be nice--probably just "1" and "2" is fine. Also I typically like defining an inline helper which takes in something like a summary and an id and gives back the DDS you want (should be a bit shorter than the copy paste, and the semantics are easy enough to understand)